ONSdigital/ras-collection-instrument
# Start of migration work

# What and why?

Currently new instruments are saved to the bucket, but older ones are still in the database.  This PR adds an endpoint to get the data, decrypt it, then upload it to the bucket.  It then sets the gcs flag to true against the instrument so this service can know where to find it.

# How to test?

- Deploy this PR to your environment
- Edit the deployment in k8s and update SEFT_GCS_ENABLED to false.  This will save any set up collection instruments to the database instead of the bucket.
- Run the SEFT acceptance tests, verify the data is in the database and the gcs flag is false.
- Find the instrument id for the instrument you want to migrate
- Add the id to the migration script in this PR.  Then run it by doing `pipenv run python migrate_instruments_to_gcp.py`
-  It should work and you should be able to see it in gcs in your namespace.  Also the gcs flag should be true against the instrument in question.
- Try and download the SEFT file and open it as a respondent.

Note the data is still in the database after this process.  Once we've migrated them all to the bucket, and left it a few days to verify that everything is okay, we can run some SQL to delete all the `data` values from the database.


# Trello
